lib/combineNodes.js

```diff
@@ -0,0 +1,310 @@
+'use strict';
+var arrayEqual = require('array-equal');
+var Cesium = require('cesium');
+var clone = Cesium.clone;
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var Matrix4 = Cesium.Matrix4;
+var Cartesian3 = Cesium.Cartesian3;
+var combineMeshes = require('./combineMeshes');
+var combinePrimitives = require('./combinePrimitives');
+module.exports = combineNodes;
+
+//Combines nodes which are not targeted for animation and share the same material. 
+//Should be run after the combineMeshes and combinePrimitives stages, giving nodes one mesh each.
+function combineNodes(gltf, preserveCameras) {
+    var animations = gltf.animations;

```

Comment:
So V8 knows more about the type and has the potential to optimize better, perhaps add `preserveCameras = defaultValue(preserveCameras, false);` instead of relying on `undefined` to implicitly convert to `false`.